### DISCUSSION TITLE
*SOLVED* firmware not actually updating

### DISCUSSION THREAD

**Discussion by brcisna** [danielewood/sierra-wireless-modems#93] [Q&A]
Hello All,

EDIT2: *SOLVED*
After running many many at commands trying to rest to factory default settings etc,the em7455 finally updated manually to 2.20.##,,,then run the autouflash-7455 and this time it did update to latest firmware.

EDIT: Adding,,(After below autoflash failed) 
Also tried doing a manual firmware up date via the 'Flash using Sierra Wireless Linux Flashing Tool (fwdwl-lite)'
This appears to flash successfully but the firmware on modem remians the same.
Next monday going to rig a windows machine and try an EXE as much as i hate to.

After what appeared to be a successful flash could not get the modem to connect to an activated sim / carrier.
After some looking the firmware is actually at the original version of -02.08.02.00. Seeing how this is my first messing with this setup didnt even think to look at firmware version.
Will attach the console output below. If you look there are two instances were the script claims cat: /dev/ttyUSB2: No such file or directory ,,and later cat: /dev/ttyUSB3: No such file or directory. Am guessing this is cause of failing even though the end of script reports success!

`UUID::Tiny is up to date (1.04).
IPC::Shareable is up to date (1.06).
JSON is up to date (4.03).
Waiting for modem to reboot...
---
Reseting modem...
Setting MBIM Mode for Modem
Interface bitmask: 0000100D (diag,nmea,modem,mbim)
---
Running Modem Mode Switch to usbcomp=8
Running in MBIM mode (driver=cdc_mbim)
MBIM OPEN succeeded
QMI msg '0x0021' returned status = 1
MBIM QMI support verified
supports 32 QMI subsystems:
  0x00 (1.5)	'QMI_CTL'	- Control service
  0x01 (1.67)	'QMI_WDS'	- Wireless data service
  0x02 (1.14)	'QMI_DMS'	- Device management service
  0x03 (1.25)	'QMI_NAS'	- Network access service
  0x04 (1.6)	'QMI_QOS'	- Quality of service, err, service 
  0x05 (1.10)	'QMI_WMS'	- Wireless messaging service
  0x07 (1.3)	'QMI_AUTH'	- Authentication service
  0x08 (1.2)	'QMI_AT'	- AT command processor service
  0x09 (2.1)	'QMI_VOICE'	- Voice service
  0x0a (2.24)	'QMI_CAT2'	- Card application toolkit service (new)
  0x0b (1.45)	'QMI_UIM'	- UIM service
  0x0c (1.4)	'QMI_PBM'	- Phonebook service
  0x0f (1.0)	'QMI_TEST'	- Test service
  0x10 (2.0)	'QMI_LOC'	- Location service 
  0x11 (1.0)	'QMI_SAR'	- Specific absorption rate service
  0x17 (1.0)	'QMI_TS'	- Thermal sensors service
  0x18 (1.0)	'QMI_TMD'	- Thermal mitigation device service
  0x1a (1.16)	'QMI_WDA'	- Wireless data administrative service
  0x1d (1.1)	'QMI_CSVT'	- Circuit switched videotelephony service
  0x22 (1.0)	'QMI_COEX'	- Coexistence service
  0x24 (1.0)	'QMI_PDC'	- Persistent device configuration service
  0x29 (1.0)	'QMI_RFRPE'	- RF radiated performance enhancement service
  0x2a (1.0)	'QMI_DSD'	- Data system determination service
  0x2b (1.0)	'QMI_SSCTL'	- Subsystem control service
  0x2e (1.0)	'unknown'	- 
  0x30 (1.0)	'unknown'	- 
  0x31 (1.0)	'unknown'	- 
  0x36 (1.0)	'unknown'	- 
  0xe1 (1.0)	'QMI_RMS'	- Remote management service
  0xf0 (1.0)	'unknown'	- 
  0xf5 (1.0)	'unknown'	- 
  0xf6 (1.0)	'unknown'	- 
QMI msg '0x0022' returned status = 1
Got QMI DMS client ID '3'
QMI msg '0x555b' returned status = 1
Current USB composition: 8
USB compositions:
   0 - HIP  DM    NMEA  AT    MDM1  MDM2  MDM3  MS      NOT SUPPORTED
   1 - HIP  DM    NMEA  AT    MDM1  MS                  NOT SUPPORTED
   2 - HIP  DM    NMEA  AT    NIC1  MS                  NOT SUPPORTED
   3 - HIP  DM    NMEA  AT    MDM1  NIC1  MS            NOT SUPPORTED
   4 - HIP  DM    NMEA  AT    NIC1  NIC2  NIC3  MS      NOT SUPPORTED
   5 - HIP  DM    NMEA  AT    ECM1  MS                  NOT SUPPORTED
   6 - DM   NMEA  AT    QMI                             SUPPORTED
   7 - DM   NMEA  AT    RMNET1 RMNET2 RMNET3            NOT SUPPORTED
*  8 - DM   NMEA  AT    MBIM                            SUPPORTED
   9 - MBIM                                             SUPPORTED
  10 - NMEA MBIM                                        NOT SUPPORTED
  11 - DM   MBIM                                        NOT SUPPORTED
  12 - DM   NMEA  MBIM                                  NOT SUPPORTED
  13 - Config1: comp6    Config2: comp8                 NOT SUPPORTED
  14 - Config1: comp6    Config2: comp9                 NOT SUPPORTED
  15 - Config1: comp6    Config2: comp10                NOT SUPPORTED
  16 - Config1: comp6    Config2: comp11                NOT SUPPORTED
  17 - Config1: comp6    Config2: comp12                NOT SUPPORTED
  18 - Config1: comp7    Config2: comp8                 NOT SUPPORTED
  19 - Config1: comp7    Config2: comp9                 NOT SUPPORTED
  20 - Config1: comp7    Config2: comp10                NOT SUPPORTED
  21 - Config1: comp7    Config2: comp11                NOT SUPPORTED
  22 - Config1: comp7    Config2: comp12                NOT SUPPORTED
Current setting is already '8'
QMI msg '0x0023' returned status = 1
Waiting for modem to reboot...
---
Reseting modem...

NO CARRIER

OK

OK

OK
Waiting for modem to reboot...
Waiting for modem to reboot...
---
Current modem settings:

OK

OK
ATI
Manufacturer: Sierra Wireless, Incorporated
Model: EM7455
Revision: SWI9X30C_02.08.02.00 r5428 CARMD-EV-FRMWR2 2016/01/06 20:38:53
MEID: 35399007578716
IMEI: 353990075787163
IMEI SV:  4
FSN: LF712120610110
+GCAP: +CGSM

OK
AT!ENTERCND="A710"
OK
AT!IMPREF?
!IMPREF: 
 preferred fw version:    02.08.02.00
 preferred carrier name:  ATT
 preferred config name:   ATT_002.009_001
 current fw version:      02.08.02.00
 current carrier name:    ATT
 current config name:     ATT_002.009_001

OK
AT!GOBIIMPREF?
!GOBIIMPREF: 
 preferred fw version:    02.08.02.00
 preferred carrier name:  ATT
 preferred config name:   ATT_002.009_001
 current fw version:      02.08.02.00
 current carrier name:    ATT
 current config name:     ATT_002.009_001

OK
AT!USBSPEED?
ERROR
cat: /dev/ttyUSB2: No such file or directory
Waiting for modem to reboot...
Downloading https://download.ttl.one/swi/7455/SWI9X30C_02.33.03.00_GENERIC_002.072_000.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 60.7M  100 60.7M    0     0   171k      0  0:06:03  0:06:03 --:--:--  173k
Archive:  SWI9X30C_02.33.03.00_GENERIC_002.072_000.zip
  inflating: SWI9X30C_02.33.03.00.cwe  
  inflating: SWI9X30C_02.33.03.00_GENERIC_002.072_000.nvu  
---
Flashing SWI9X30C_02.33.03.00.cwe onto Generic Sierra Modem...
reseter operation finished successfully
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Waiting for modem in boothold mode...
Found 413c:81b5
downloading cwe image: SWI9X30C_02.33.03.00.cwe (64.4 MB)...
finalizing download... (may take more than one minute, be patient)
successfully downloaded in 57.22s (1.1 MB/s)
downloading cwe image: SWI9X30C_02.33.03.00_GENERIC_002.072_000.nvu (3.3 kB)...
finalizing download... (may take more than one minute, be patient)
successfully downloaded in 0.05s (62.9 kB/s)
rebooting in normal mode...
firmware update operation finished successfully
cat: /dev/ttyUSB3: No such file or directory
Done!
`

**Comment by corty8** [comment]
Hi there

I am having this same problem but I am unsure exactly what you did to solve it, the script runs through in its entirety and says its successful but the it still shows up as a Dell Device no the EM7455

Also at the end of the script I also get the  cat: /dev/ttyUSB3: No such file or directory

Any help would be appreciated

Cheers

**Reply by brcisna** [reply]
@corty8 rty8

Hi Corty8
Something to keep in mind this device will always show up in AT commands and USB interface Dell Device. This isn't a big deal.
Am still very green at this ,myself,,but have spent MANY hours trying to learn the AT commands to see what is going on inside the modem,
I actually wound up using the Windows exe Generic Firmware,to get my first EM7455 to work, Looking back now,,I had no idea what I should be seeing as a successful flash.
My suggestion would be to try a Windows machine and download the Generic Firmware from the Sierra Wireless site.version 2.33.03.## 
If you still have troubles,,,would be glad to help you remotely,,it would cut down on the time you spend trying to get it to work,most likely
Do you have a M.2 to USB adapter I assume? If you have anydesk,we could give it a shot,Its up to you. You are working in a linux box?
Good Luck.

**Reply by corty8** [reply]
Hi Brcisna

Thanks for your reply

I actually tried downloading the generic exe firmware version yesterday after my message but it did not seem to do anything, it runs through the process but on completion it has not changed the modem and still reports as a Dell part number

Yes I have the modem in an m2 USB device and running on windows

I am guessing when I look at Daniels script there is a number of AT commands that need to happen before the firmware can be changed 

I actually have 2 em7455 and also some em7430, it is the em7430 that I need to do the most at the moment


Cheers

**Reply by brcisna** [reply]
@corty8 

Not to change the subject, but have you ever figured out a good solution to get around CGNAT. Myself,was not at all familiar with cgnat but getting into this cellular internet thing,,and buying a helium miner,about the same time, ,just to play around with,,has really been a  challenge to get a  solid working solution.
Have been messing with Wireguard for about two weeks now,,off and on, and is very sketchy at best, It is fairly easy to setup,but not sure what is happening,it seems like dns,or the code itself,,is just very dodgey. I am probably doing something wrong. The browsing speed with it enabled is almost unusable for me.

**Reply by danielewood** [reply]
> Not to change the subject, but have you ever figured out a good solution to get around CGNAT. Myself,was not at all familiar with cgnat but getting into this cellular internet thing,,and buying a helium miner,about the same time, ,just to play around with,,has really been a challenge to get a solid working solution.

Two things, setting your Wireguard MTU to 1360 on both the client and server may help. But it may just be too much packet loss on the link for reliable UDP communications. You should also try OpenVPN over TCP and see if that fares better.

**Comment by danielewood** [comment]
> OK
AT!USBSPEED?
ERROR
cat: /dev/ttyUSB2: No such file or directory

This indicates you modem disconnected immediately after ERROR.

You can find that here:
https://github.com/danielewood/sierra-wireless-modems/blob/50fe059fab83fa591fcc27c4b6e8518ede4eee68/autoflash-7455.sh#L252

You can run with:
```bash
sudo bash autoflash-7455.sh -s
```
to only change the modem settings, which will change the PID/VID.

Manually you can accomplish the same thing by running:
```
ATE1
AT!ENTERCND="A710"
AT!USBVID=1199
AT!USBPID=9071,9070
AT!USBPRODUCT="EM7455"
AT!RESET
```

That will cause the modem to identify as a Generic Sierra modem, and then the Sierra Windows flasher will recognize it.


@corty8 Ignore what I said a few months ago about the 7430 being different, I misread that as 7340. The 7430 uses the same everything, its just different band supports for the same hardware as the 7455.
